presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java

```diff
@@ -188,29 +194,32 @@ private boolean isAggregationPushdownSupported(AggregationNode partialAggregatio
             return true;
         }
 
-        private boolean isSupportedOrcTypeForMinMax(Type type)
+        private boolean isNotSupportedOrcTypeForMinMax(Type type)
         {
-            return !BOOLEAN.equals(type) &&
-                    type.getJavaType() != boolean.class &&
-                    !isRowType(type) &&
-                    !isArrayType(type) &&
-                    !isMapType(type) &&
-                    !TINYINT.equals(type) &&
-                    !VARBINARY.equals(type) &&
-                    !TIMESTAMP.equals(type) &&
-                    ((!VARBINARY.equals(type) && !VARCHAR.equals(type)) ||
-                            isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session));
+            return BOOLEAN.equals(type) ||
+                    type.getJavaType() == boolean.class ||
+                    isRowType(type) ||
+                    isArrayType(type) ||
+                    isMapType(type) ||
+                    TINYINT.equals(type) ||
+                    VARBINARY.equals(type) ||
+                    TIMESTAMP.equals(type) ||
+                    isNotSupportedForVariableLengthDataType(type);
         }
 
-        private boolean isSupportedParquetTypeForMinMax(Type type)
+        private boolean isNotSupportedParquetTypeForMinMax(Type type)
         {
-            return !BOOLEAN.equals(type) &&
-                    type.getJavaType() != boolean.class &&
-                    !isRowType(type) &&
-                    !isArrayType(type) &&
-                    !isMapType(type) &&
-                    ((!VARBINARY.equals(type) && !VARCHAR.equals(type)) ||
-                            isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session));
+            return BOOLEAN.equals(type) ||
+                    type.getJavaType() == boolean.class ||
+                    isRowType(type) ||
+                    isArrayType(type) ||
+                    isMapType(type) ||
+                    isNotSupportedForVariableLengthDataType(type);
+        }
+
+        private boolean isNotSupportedForVariableLengthDataType(Type type)
+        {
+            return VARCHAR.equals(type) && !isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session);
```

Comment:
does this mean parquet is no longer checking this session property for varbinary fields?

Reply:
Looks like parquet needs it. Test failure is related. cc @sdruzkin

Reply:
Added a separate check for parquet which includes varbinary

Reply:
Tests are passing now. Can you take another look.